shell/common/switches.cc

```diff
@@ -36,6 +37,22 @@ struct SwitchDesc {
 #define DEF_SWITCHES_END };
 // clang-format on
 
+#if FLUTTER_RUNTIME_MODE != FLUTTER_RUNTIME_MODE_RELEASE && \
+    FLUTTER_RUNTIME_MODE != FLUTTER_RUNTIME_MODE_DYNAMIC_RELEASE
+
+// List of common and safe VM flags to allow to be passed directly to the VM.
+static const std::string gDartFlagsWhitelist[] = {
```

Comment:
The current whitelist is just composed of some flags that I thought would be useful and safe. I'm happy to add/remove from this list based on feedback.